tests/postgres_tests/test_indexes.py

```diff
@@ -132,6 +134,46 @@ def test_deconstruction_no_customization(self):
         self.assertEqual(kwargs, {'fields': ['title'], 'name': 'test_title_gist'})
 
 
+@skipUnlessDBFeature('has_hash_index_support')
+class HashIndexTests(PostgreSQLTestCase):
+
+    def test_suffix(self):
+        self.assertEqual(HashIndex.suffix, 'hash')
+
+    def test_eq(self):
+        index = HashIndex(fields=['title'], fillfactor=64)
+        same_index = HashIndex(fields=['title'], fillfactor=64)
+        another_index = HashIndex(fields=['author'])
+        self.assertEqual(index, same_index)
+        self.assertNotEqual(index, another_index)
+
+    def test_name_auto_generation(self):
+        index = HashIndex(fields=['field'])
+        index.set_name_with_model(CharFieldModel)
+        self.assertEqual(index.name, 'postgres_te_field_baaf2b_hash')
+
+    def test_deconstruction(self):
+        index = HashIndex(fields=['title'], name='test_title_hash', fillfactor=80)
+        path, args, kwargs = index.deconstruct()
+        self.assertEqual(path, 'django.contrib.postgres.indexes.HashIndex')
+        self.assertEqual(args, ())
+        self.assertEqual(
+            kwargs,
+            {
+                'fields': ['title'],
+                'name': 'test_title_hash',
+                'fillfactor': 80,
+            }
+        )
+
+    def test_deconstruction_no_customization(self):
```

Comment:
As above -- it seems somewhat useless to implement this test for every subclass. At the least, maybe there could be a base test class that implements some common tests if that's warranted.

Reply:
Agreed. Will address this and the previous together.